sumpy/expansion/__init__.py

```diff
@@ -211,228 +226,280 @@ def get_full_kernel_derivatives_from_stored(self, stored_kernel_derivatives,
 
 # {{{ sparse matrix-vector multiplication
 
-def _spmv(spmat, x, sparse_vectors):
+class CSEMatVecOperator(object):
     """
-    :param spmat: maps row indices to list of (col idx, value)
-    :param x: maps vector indices to values
-    :param sparse_vectors: If True, treat vectors as dict-like, otherwise list-like
+    A class to facilitate a fast matrix vector multiplication with
+    common subexpression eliminated. In compressed Taylor
+    series, the compression matrix's operation count can be
+    reduced to `O(p**d)` from `O(p**{2d-1})` using CSE.
+
+    .. attribute:: assignments
```

Comment:
* I don't find the name `assignments` to be super descriptive of what's inside.
* What do you think of separating this into two lists? Opening bid on naming: `from_input_coeffs_by_row` and `from_output_coeffs_by_row`?

Reply:
Sure. Separating into two lists is fine. Not a fan of the names, but couldn't think of better names.